IdentityCore/src/webview/embeddedWebview/challangeHandlers/mac/MSIDCertAuthHandler.m

```diff
@@ -72,7 +76,16 @@ + (BOOL)handleChallenge:(NSURLAuthenticationChallenge *)challenge
                  completionHandler(NSURLSessionAuthChallengeRejectProtectionSpace, nil);
                  return;
              }
-             
+            
+            // If there is no preferred identity saved, we must set preferred identity certificate using hostname and key usage parameters
+            if (identity == NULL)
+            {
+                NSArray *arr = @[(__bridge NSString *)kSecAttrCanSign, (__bridge NSString *)kSecAttrCanEncrypt, (__bridge NSString *)kSecAttrCanDecrypt];
+                CFArrayRef arrayRef = (__bridge CFArrayRef)arr;
+                OSStatus status = SecIdentitySetPreferred(selectedIdentity, (CFStringRef)host, arrayRef);
```

Comment:
Just curious: Set identity here declares list of key usages but the get query above these key usages is null, should they be the same?

Reply:
I don't think they need to be the same because when we are getting the query, we just want to make sure we find any available cert whereas when we're setting it, we're putting the default attributes (the main use cases that are recommended)